pkg/jobs/job.go

```diff
@@ -339,6 +349,23 @@ func (job *Job) teardownWithNoCallbacks(result string) error {
 	return nil
 }
 
+func (job *Job) uploadLogsAsArtifact(filePath string) {
+	if !job.UploadTrimmedLogs {
+		log.Infof("Logs were trimmed, but agent is not configured to upload them as artifact - skipping.")
+		return
+	}
+
+	log.Infof("Uploading job logs as artifact...")
+	cmd := []string{"artifact", "push", "job", filePath, "-d", "agent/job_logs.json"}
```

Comment:
The official format of the job logs is JSONL https://jsonlines.org.
If we want to be ultra-mathematically precise, we should use the `.jsonl` extension.

---

Contra argument: JSONL is a relatively unknown format, it might just make it confusing if it.

---

I don't have a hard stance on this; just leaving my thoughts.

Reply:
I think `.json` might be better because it might be confusing to see the logs as a `.jsonl`. Although it might be confusing to see the logs as a JSON file as well, so another alternative might be to pre-process that file and upload the logs as plain text (just like the raw logs endpoint in the UI).

Reply:
Preprocessing is a good idea and a reminder that those JSON files are impossible to read because every entry is a buffer flush, not a complete line.

I'm 90% sure this will be requested after the first actual usage.

Reply:
I'll do that then.